### ISSUE TITLE
Adding a column specified in Foreign Key to blacklist causes panic when generating code

### ISSUE THREAD

**Issue by touyu** [volatiletech/sqlboiler#695]
### What version of SQLBoiler are you using (`sqlboiler --version`)?
SQLBoiler v3.6.1

### What is your database and version (eg. Postgresql 10)
mysql  Ver 8.0.18 for osx10.14 on x86_64 (Homebrew)

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)
sqlboiler --wipe mysql

### What is the output of the command above with the `-d` flag added to it? (Provided you are comfortable sharing this, it contains a blueprint of your schema)
```
{
    "config": {
        "driver_name": "mysql",
        "driver_config": {
            "blacklist": [
                "items.user_id"
            ],
            "dbname": "sqlboiler",
            "host": "127.0.0.1",
            "port": 3306,
            "sslmode": "false",
            "user": "root",
            "whitelist": null
        },
        "pkg_name": "models",
        "out_folder": "models",
        "debug": true,
        "wipe": true,
        "struct_tag_casing": "snake",
        "imports": {
            "all": {
                "Standard": [
                    "\"database/sql\"",
                    "\"fmt\"",
                    "\"reflect\"",
                    "\"strings\"",
                    "\"sync\"",
                    "\"time\""
                ],
                "ThirdParty": [
                    "\"github.com/friendsofgo/errors\"",
                    "\"github.com/volatiletech/sqlboiler/boil\"",
                    "\"github.com/volatiletech/sqlboiler/queries\"",
                    "\"github.com/volatiletech/sqlboiler/queries/qm\"",
                    "\"github.com/volatiletech/sqlboiler/queries/qmhelper\"",
                    "\"github.com/volatiletech/sqlboiler/strmangle\""
                ]
            },
            "test": {
                "Standard": [
                    "\"bytes\"",
                    "\"reflect\"",
                    "\"testing\""
                ],
                "ThirdParty": [
                    "\"github.com/volatiletech/sqlboiler/boil\"",
                    "\"github.com/volatiletech/sqlboiler/queries\"",
                    "\"github.com/volatiletech/sqlboiler/randomize\"",
                    "\"github.com/volatiletech/sqlboiler/strmangle\""
                ]
            },
            "singleton": {
                "boil_queries": {
                    "Standard": null,
                    "ThirdParty": [
                        "\"github.com/volatiletech/sqlboiler/drivers\"",
                        "\"github.com/volatiletech/sqlboiler/queries\"",
                        "\"github.com/volatiletech/sqlboiler/queries/qm\""
                    ]
                },
                "boil_types": {
                    "Standard": [
                        "\"strconv\""
                    ],
                    "ThirdParty": [
                        "\"github.com/friendsofgo/errors\"",
                        "\"github.com/volatiletech/sqlboiler/boil\"",
                        "\"github.com/volatiletech/sqlboiler/strmangle\""
                    ]
                }
            },
            "test_singleton": {
                "boil_main_test": {
                    "Standard": [
                        "\"database/sql\"",
                        "\"flag\"",
                        "\"fmt\"",
                        "\"math/rand\"",
                        "\"os\"",
                        "\"path/filepath\"",
                        "\"strings\"",
                        "\"testing\"",
                        "\"time\""
                    ],
                    "ThirdParty": [
                        "\"github.com/spf13/viper\"",
                        "\"github.com/volatiletech/sqlboiler/boil\""
                    ]
                },
                "boil_queries_test": {
                    "Standard": [
                        "\"bytes\"",
                        "\"fmt\"",
                        "\"io\"",
                        "\"io/ioutil\"",
                        "\"math/rand\"",
                        "\"regexp\""
                    ],
                    "ThirdParty": [
                        "\"github.com/volatiletech/sqlboiler/boil\""
                    ]
                },
                "boil_suites_test": {
                    "Standard": [
                        "\"testing\""
                    ],
                    "ThirdParty": null
                }
            }
        },
        "aliases": {},
        "version": "3.6.1"
    },
    "driver_config": {
        "blacklist": [
            "items.user_id"
        ],
        "dbname": "sqlboiler",
        "host": "127.0.0.1",
        "port": 3306,
        "sslmode": "false",
        "user": "root",
        "whitelist": null
    },
    "schema": "",
    "dialect": {
        "lq": 0,
        "rq": 0,
        "use_index_placeholders": false,
        "use_last_insert_id": false,
        "use_schema": false,
        "use_default_keyword": false,
        "use_auto_columns": false,
        "use_top_clause": false,
        "use_output_clause": false,
        "use_case_when_exists_clause": false
    },
    "tables": null,
    "templates": null
}
```

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)
```
CREATE TABLE `users` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `name` int(11) DEFAULT NULL,
  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;

CREATE TABLE `items` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `user_id` int(11) unsigned NOT NULL,
  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  KEY `items_ibfk_1` (`user_id`),
  CONSTRAINT `items_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
```

### Further information. What did you do, what did you expect?
I expected that adding `items.user_id` to the blacklist would remove `user_id` from `items` and remove `items` from `users`' relations.

However running `sqlboiler --wipe mysql` caused panic
```
panic: could not find column name: user_id [recovered]
panic: interface conversion: string is not error: missing method Error
```

Is there a way to solve this problem?

**Comment by aarondl** [comment]
The code could probably be adjusted to make this use case work, but I don't think there's any way around it currently.

**Comment by stephenafamo** [comment]
Fixed by #1055
